pxtlib/webusb.ts

```diff
@@ -421,104 +447,101 @@ namespace pxt.usb {
-                    let iface = matching[matching.length - 1]
-                    this.log(`${matching.length} matching interfaces; picking ${iface ? "#" + iface.interfaceNumber : "n/a"}`)
-                    if (!iface)
-                        this.error("cannot find supported USB interface")
-                    this.altIface = iface.alternates[0]
-                    this.iface = iface
-                    if (this.altIface.endpoints.length) {
-                        this.log("using dedicated endpoints")
-                        this.epIn = this.altIface.endpoints.filter(e => e.direction == "in")[0]
-                        this.epOut = this.altIface.endpoints.filter(e => e.direction == "out")[0]
-                        Util.assert(this.epIn.packetSize == 64);
-                        Util.assert(this.epOut.packetSize == 64);
-                    } else {
-                        this.log("using ctrl pipe")
                     }
-                    this.log("claim interface")
-                    return dev.claimInterface(iface.interfaceNumber)
-                })
-                .then(() => {
-                    this.log("device ready")
-                    this.lastKnownDeviceSerialNumber = this.dev.serialNumber;
-                    this.ready = true;
-                    if (isHF2)
-                        this.readLoop();
-                    if (this.onConnectionChanged)
-                        this.onConnectionChanged();
-                })
+                }
+                return false;
+            }
+            this.log("got " + dev.configurations[0].interfaces.length + " interfaces");
+            const matching = dev.configurations[0].interfaces.filter(matchesFilters);
+            let iface = matching[matching.length - 1];
+            this.log(`${matching.length} matching interfaces; picking ${iface ? "#" + iface.interfaceNumber : "n/a"}`);
+            if (!iface)
+                this.error("cannot find supported USB interface");
+            this.altIface = iface.alternates[0];
+            this.iface = iface;
+            if (this.altIface.endpoints.length) {
+                this.log("using dedicated endpoints");
+                this.epIn = this.altIface.endpoints.filter(e => e.direction == "in")[0];
+                this.epOut = this.altIface.endpoints.filter(e => e.direction == "out")[0];
+                Util.assert(this.epIn.packetSize == 64);
+                Util.assert(this.epOut.packetSize == 64);
+            } else {
+                this.log("using ctrl pipe");
+            }
+            this.log("claim interface");
+            await dev.claimInterface(iface.interfaceNumber);
+            this.log("device ready");
+            this.lastKnownDeviceSerialNumber = this.dev.serialNumber;
+            this.ready = true;
+            if (isHF2) {
+                // just starting, not waiting on it.
+                this.readLoop();
```

Comment:
For extra clarity on intent here:
```suggestion
                /*await*/ this.readLoop();
```